airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/products_report.py

```diff
@@ -243,6 +243,17 @@
 }
 
 
+METRICS_TYPE_TO__MAP = {
+    "campaigns": "campaignId",
+    "adGroups": "adGroupId",
+    "keywords": "keywordId",
+    "productAds": "adId",
+    "asins_keywords": "asin",
+    "asins_targets": "asin",
+    "targets": "targetId",
+}
```

Comment:
As we can see there is an abstract `metrics_map` property that must be implemented by every successor of the `ReportStream` class. Now we need to extend it by adding an id mapping for every `metrics_map`. If we put this mapping to the base class then it turns out that we put implementation to the base class, but it tied to implementation of its successors. Then we need to combine all the `metrics_map` and also move them to the base class... It seems not correct to me. I like the idea that it is better to extend a logic instead of making some kind of god object (at least in this case). And last but not least sometimes key/values pairs are differ:
- products report: `"keywords": "keywordId"`
- brands video report: `"keywords": "keywordBid"`

Reply:
Got it - in the case that they sometimes differ, I think your implementation makes sense. I wasn't sure if the metrics_map was also shared between all the subclasses as well and could be combined into the base class. Thanks for the clarification :)